sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -26,15 +26,23 @@ public void Execute(GeneratorExecutionContext context)
 
             // attempt to parse user compilation
             var p = new Parser(context);
-            IReadOnlyList<GeneratorFunctionMetadata> functionMetadataInfo = p.GetFunctionMetadataInfo(receiver.CandidateMethods);
 
-            // Proceed to generate the file if function metadata info was successfully returned
-            if (functionMetadataInfo.Count > 0)
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.FunctionsDisableFuncMetadataSourceGen", out var sourceGenSwitch);
+
+            bool.TryParse(sourceGenSwitch, out bool disableSourceGen);
+
+            if (!disableSourceGen)
```

Comment:
Nit: do we want to check for disableSourceGen == true and return early, so that we can move logic out of the `{ }` similar to the patten on line 22